src/Microsoft.SqlTools.ServiceLayer/DacFx/ValidateStreamingJobOperation.cs

```diff
@@ -0,0 +1,123 @@
+{
+    /// <summary>
+    /// Class to represent an in-progress deploy operation
+    /// </summary>
+    class ValidateStreamingJobOperation
+    {
+        public ValidateStreamingJobParams Parameters { get; }
+
+        public ValidateStreamingJobOperation(ValidateStreamingJobParams parameters)
+        {
+            Validate.IsNotNull("parameters", parameters);
+            this.Parameters = parameters;
+        }
+
+        public ValidateStreamingJobResult ValidateQuery()
+        {
+            try
+            {
+                TSqlModel model = TSqlModel.LoadFromDacpac(Parameters.PackageFilePath, new ModelLoadOptions(SqlServer.Dac.DacSchemaModelStorageType.Memory, loadAsScriptBackedModel: true));
+
+                (string name, string statement) = ExtractStreamingJobData(Parameters.CreateStreamingJobTsql);
+                ASA::ParseResult referencedStreams = ParseStatement(statement);
+
+                List<TSqlObject> streams = model.GetObjects(DacQueryScopes.Default, ExternalStream.TypeClass).ToList();
+                HashSet<string> identifiers = streams.Select(x => x.Name.Parts[^1]).ToHashSet();
+
+                List<string> errors = new List<string>();
+
+                foreach (ASA::SchemaObjectName stream in referencedStreams.Inputs.Values)
+                {
+                    if (!identifiers.Contains(stream.BaseIdentifier.Value))
+                    {
+                        errors.Add(SR.StreamNotFoundInModel(SR.Input, stream.BaseIdentifier.Value));
+                    }
+                }
+
+                foreach (ASA::SchemaObjectName stream in referencedStreams.Outputs.Values)
+                {
+                    if (!identifiers.Contains(stream.BaseIdentifier.Value))
+                    {
+                        errors.Add(SR.StreamNotFoundInModel(SR.Output, stream.BaseIdentifier.Value));
+                    }
+                }
+
+                return new ValidateStreamingJobResult()
+                {
+                    Success = errors.Count == 0,
+                    ErrorMessage = errors.Count == 0 ? null : SR.StreamingJobValidationFailed(name) + Environment.NewLine + String.Join(Environment.NewLine, errors)
+                };
+            }
+            catch (Exception ex)
+            {
+                return new ValidateStreamingJobResult()
+                {
+                    Success = false,
+                    ErrorMessage = ex.Message
+                };
+            }
+        }
+
+        private (string JobName, string JobStatement) ExtractStreamingJobData(string createStreamingJobTsql)
+        {
+            TSqlParser parser = new TSql150Parser(initialQuotedIdentifiers: true);
+
+            TSqlFragment fragment = parser.Parse(new StringReader(createStreamingJobTsql), out IList<ParseError> errors);
+
+            if (((TSqlScript)fragment).Batches.Count != 1)
+            {
+                throw new ArgumentException("TSQL fragment should contain exactly one batch.");
```

Comment:
On second thought, probably should.  I originally didn't because they're bad paths that the user shouldn't be getting themselves into, but they can find their way here without doing anything too hacky.